tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -189,6 +189,154 @@ public void SetParameters(TestParameters parameters)
         }
     }
 
+    /// <summary>
+    /// Set benchmark metadata
+    /// </summary>
+    /// <param name="hostInfo">Host info</param>
+    /// <param name="jobInfo">Job info</param>
+    public void SetBenchmarkMetadata(BenchmarkHostInfo hostInfo, BenchmarkJobInfo jobInfo)
+    {
+        ((TestSpanTags)_scope.Span.Tags).Type = TestTags.TypeBenchmark;
+
+        // Host info
+
+        if (hostInfo.ProcessorName is not null)
+        {
+            SetTag(BenchmarkTestTags.HostProcessorName, hostInfo.ProcessorName);
+        }
+
+        if (hostInfo.ProcessorCount is not null)
+        {
+            SetTag(BenchmarkTestTags.HostProcessorPhysicalProcessorCount, hostInfo.ProcessorCount);
+        }
+
+        if (hostInfo.PhysicalCoreCount is not null)
+        {
+            SetTag(BenchmarkTestTags.HostProcessorPhysicalCoreCount, hostInfo.PhysicalCoreCount);
+        }
+
+        if (hostInfo.LogicalCoreCount is not null)
+        {
+            SetTag(BenchmarkTestTags.HostProcessorLogicalCoreCount, hostInfo.LogicalCoreCount);
+        }
+
+        if (hostInfo.ProcessorMaxFrequencyHertz is not null)
+        {
+            SetTag(BenchmarkTestTags.HostProcessorMaxFrequencyHertz, hostInfo.ProcessorMaxFrequencyHertz);
+        }
+
+        if (hostInfo.OsVersion is not null)
+        {
+            SetTag(BenchmarkTestTags.HostOsVersion, hostInfo.OsVersion);
+        }
+
+        if (hostInfo.RuntimeVersion is not null)
+        {
+            SetTag(BenchmarkTestTags.HostRuntimeVersion, hostInfo.RuntimeVersion);
+        }
+
+        if (hostInfo.ChronometerFrequencyHertz is not null)
+        {
+            SetTag(BenchmarkTestTags.HostChronometerFrequencyHertz, hostInfo.ChronometerFrequencyHertz);
+        }
+
+        if (hostInfo.ChronometerResolution is not null)
+        {
+            SetTag(BenchmarkTestTags.HostChronometerResolution, hostInfo.ChronometerResolution);
+        }
+
+        // Job info
+
+        if (jobInfo.Description is not null)
+        {
+            SetTag(BenchmarkTestTags.JobDescription, jobInfo.Description);
+        }
+
+        if (jobInfo.Platform is not null)
+        {
+            SetTag(BenchmarkTestTags.JobPlatform, jobInfo.Platform);
+        }
+
+        if (jobInfo.RuntimeName is not null)
+        {
+            SetTag(BenchmarkTestTags.JobRuntimeName, jobInfo.RuntimeName);
+        }
+
+        if (jobInfo.RuntimeMoniker is not null)
+        {
+            SetTag(BenchmarkTestTags.JobRuntimeMoniker, jobInfo.RuntimeMoniker);
+        }
+    }
+
+    /// <summary>
+    /// Add benchmark data
+    /// </summary>
+    /// <param name="measureType">Measure type</param>
+    /// <param name="info">Measure info</param>
+    /// <param name="statistics">Statistics values</param>
+    public void AddBenchmarkData(BenchmarkMeasureType measureType, string info, BenchmarkDiscreteStats statistics)
```

Comment:
nit: `in`
```suggestion
    public void AddBenchmarkData(BenchmarkMeasureType measureType, string info, in BenchmarkDiscreteStats statistics)
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3774/commits/83258a6d8396e8ff551e7e941baccff1166d9498